flexget/components/notify/notifiers/slack.py

```diff
@@ -106,16 +255,104 @@ def notify(self, title, message, config):
         """
         Send a Slack notification
         """
-        notification = {
-            'text': message,
-            'username': config.get('username'),
-            'channel': config.get('channel'),
-            'attachments': config.get('attachments'),
-        }
-        if config.get('icon_emoji'):
-            notification['icon_emoji'] = ':%s:' % config['icon_emoji'].strip(':')
-        if config.get('icon_url'):
-            notification['icon_url'] = config['icon_url']
+
+        import json
+        logger.trace(json.dumps(self.schema))
+
+        # version 1 (default) is the Legacy formatting
+        if config.get('version') == 1:
+            # username, channel, icon_emoji, and icon_url are deprecated and ignored by the Slack API,
+            # therefore they've been removed from the posted data
+            notification = {
+                'text': message,
+                'attachments': config.get('attachments'),
+            }
+
```

Comment:
In [https://api.slack.com/legacy/custom-integrations/messaging/webhooks](https://api.slack.com/legacy/custom-integrations/messaging/webhooks), it only says “Legacy custom integrations lack newer features and will be deprecated and possibly removed in the future.” It does not say “they are ignored by the Slack API.”